internal/codeintel/codenav/gittree_translator.go

```diff
@@ -209,124 +272,123 @@ func (g *gitTreeTranslator) readHunks(ctx context.Context, repo *sgtypes.Repo, s
-	// Translate from bundle/lsp zero-index to git diff one-index
-	line = line + 1
-
-	hunk := findHunk(hunks, line)
-	if hunk == nil {
-		// Trivial case, no changes before this line
-		return line - 1, true
+func newCompactHunk(h *diff.Hunk) compactHunk {
+	// If either origLines or newLines are 0, their corresponding line is shifted by an additional -1
+	// in the `git diff` output, to make it clear to the user that the line is not included in the
+	// displayed hunk.
+	// For our purposes we need the actual start line of the hunk though
+	origStartLine := h.OrigStartLine
+	if h.OrigLines == 0 {
+		origStartLine += 1
 	}
-
-	// If the hunk ends before this line, we can simply set the line offset by the
-	// relative difference between the line offsets in each file after this hunk.
-	if line >= int(hunk.OrigStartLine+hunk.OrigLines) {
-		endOfSourceHunk := int(hunk.OrigStartLine + hunk.OrigLines)
-		endOfTargetHunk := int(hunk.NewStartLine + hunk.NewLines)
-		targetCommitLineNumber := line + (endOfTargetHunk - endOfSourceHunk)
-
-		// Translate from git diff one-index to bundle/lsp zero-index
-		return targetCommitLineNumber - 1, true
+	newStartLine := h.NewStartLine
+	if h.NewLines == 0 {
+		newStartLine += 1
 	}
+	return compactHunk{
+		origStartLine: origStartLine,
+		origLines:     h.OrigLines,
+		newStartLine:  newStartLine,
+		newLines:      h.NewLines,
+	}
+}
 
-	// These offsets start at the beginning of the hunk's delta. The following loop will
-	// process the delta line-by-line. For each line that exists the source (orig) or
-	// target (new) file, the corresponding offset will be bumped. The values of these
-	// offsets once we hit our target line will determine the relative offset between
-	// the two files.
-	sourceOffset := int(hunk.OrigStartLine)
-	targetOffset := int(hunk.NewStartLine)
-
-	for _, deltaLine := range strings.Split(string(hunk.Body), "\n") {
-		isAdded := strings.HasPrefix(deltaLine, "+")
-		isRemoved := strings.HasPrefix(deltaLine, "-")
-
-		// A line exists in the source file if it wasn't added in the delta. We set
-		// this before the next condition so that our comparison with our target line
-		// is correct.
-		if !isAdded {
-			sourceOffset++
-		}
-
-		// Hit our target line
-		if sourceOffset-1 == line {
-			// This particular line was (1) edited; (2) removed, or (3) added.
-			// If it was removed, there is nothing to point to in the target file.
-			// If it was added, then we don't have any index information for it in
-			// our source file. In any case, we won't have a precise translation.
-			if isAdded || isRemoved {
-				return 0, false
-			}
-
-			// Translate from git diff one-index to bundle/lsp zero-index
-			return targetOffset - 1, true
-		}
+func (h *compactHunk) overlapsLine(line int32) bool {
+	// git diff hunks are 1-based, vs our 0-based scip ranges
+	return h.origStartLine <= line+1 && h.origStartLine+h.origLines > line+1
```

Comment:
```suggestion
	return h.origStartLine <= line+1 && line+1 < h.origStartLine+h.origLines
```

nit: Prefer same direction for comparisons for ease of reading.